src/string/test.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import _ from "lodash";
+import { describe, it } from "mocha";
+import { expect } from "goodeggs-test-helpers";
+
+import integer from "../integer";
+import string, { removeDoubleSpace } from ".";
+
+const removeDuplicates = (str: string): string[] => _.uniq(str.split(""));
+
+describe("string", function() {
+  it("generates a random string", function() {
+    expect(string()).to.be.an("string");
+  });
+
+  it("generates a random string with legth passed-in", function() {
+    const length = integer(1);
+
+    expect(string(length).length).to.be.equal(length);
+  });
+
+  it("generates a random string with charset passed-in", function() {
+    const charset = Math.random().toString(36); // generate a random charset
+    const length = integer(100);
+    const chars = removeDuplicates(string(length, charset));
+    const charsetChars = removeDuplicates(charset);
+
+    expect(chars)
+      .to.be.an("array")
+      .that.has.length(charsetChars.length)
+      .that.have.members(charsetChars);
+  });
+
+  it("generates a random without double space and right side space", function() {
+    const charset = "a      bc ";
+    const generatedString = string(integer(100), charset);
+
+    expect(generatedString.length).to.be.equal(
+      removeDoubleSpace(generatedString).length
+    );
+  });
```

Comment:
Please add tests that exercise `fake.string`, and please also add edge cases--what if someone passes `fake.string(1|2)`?

Given that this is a random data generator, you can test this by running the function in a loop (100000 iterations should be both very fast and good enough to catch errors.)